status/reporter_gitlab.go

```diff
@@ -157,14 +158,29 @@ func (r *GitLabReporter) setCommitStatus(report TestReport) error {
 		if err != nil {
 			return fmt.Errorf("error while getting all commitStatuses for sha %s: %w", r.sha, err)
 		}
-		existingCommitStatus := r.GetExistingCommitStatus(allCommitStatuses, report.FullName)
+		existingCommitStatus = r.GetExistingCommitStatus(allCommitStatuses, report.FullName)
+
 		if existingCommitStatus != nil && existingCommitStatus.Status == string(gitlab.Running) {
 			r.logger.Info("Will not update the existing commit status from `running` to `running`",
 				"scenario.name", report.ScenarioName, "commitStatus.ID", existingCommitStatus.ID)
 			return nil
 		}
 	}
 
+	// Skip updating the commit status if the current status is "enqueue"
+	if glState == gitlab.Running || glState == gitlab.Pending {
+		allCommitStatuses, _, err := r.client.Commits.GetCommitStatuses(r.sourceProjectID, r.sha, nil)
+		if err != nil {
+			return fmt.Errorf("error while getting all commitStatuses for sha %s: %w", r.sha, err)
+		}
+		existingCommitStatus = r.GetExistingCommitStatus(allCommitStatuses, report.FullName)
```

Comment:
I think you can just use the returned `existingCommitStatus` from https://github.com/konflux-ci/integration-service/pull/853/files#diff-9173b9d26ca5486f2c07b5dfe966142cae4ca991f7787168bc520c2c655ac83aR161, no need to call it here again. HDYT?